### PR TITLE
feat: add helper text API

### PR DESCRIPTION
**Pull request by DiegoCardoso**
- Add property and named slot
- Add styles for supported themes
- Add demo for helper text API
- Add unit and visual tests for helper text

Part of #170

### CODE REVIEW

File: demo/checkbox-basic-demos.html [hunk 1]
```diff
@@ -45,6 +45,32 @@ <h3>Disabled state</h3>
       </template>
     </vaadin-demo-snippet>
 
+    <h3>Helper text</h3>
+    <p>Use the <code>helper-text</code> attribute or add content to the <code>helper</code> slot to set helper content.</p>
+    <vaadin-demo-snippet id="text-field-basic-demos-helper-text">
```
Comment (tomivirkki) [inline demo/checkbox-basic-demos.html:50]: Mentions text-field in the id. I think the component doesn't need to be mentioned in the demo id. Just "helper-text" should be enough for a demo id

File: demo/checkbox-theme-variants-demos.html [hunk 2]
```diff
@@ -24,6 +24,25 @@ <h4>Vertical</h4>
       </template>
     </vaadin-demo-snippet>
 
+    <h3>Helper text position</h3>
+    <vaadin-demo-snippet id="text-field-theme-demos-sizes">
```
Comment (tomivirkki) [inline demo/checkbox-theme-variants-demos.html:28]: The demo id isn't right

File: demo/checkbox-theme-variants-demos.html [hunk 3]
```diff
@@ -24,6 +24,25 @@ <h4>Vertical</h4>
...
+      <template preserve-content>
+        <h4>Below</h4>
+        <vaadin-checkbox-group label="Label" helper-text="Helper text below text field">
```
Comment (tomivirkki) [inline demo/checkbox-theme-variants-demos.html:31]: Mentions "text field". Maybe just "...below the field"?

File: demo/checkbox-theme-variants-demos.html [hunk 4]
```diff
@@ -24,6 +24,25 @@ <h4>Vertical</h4>
...
+          <vaadin-checkbox value="1" checked>Option one</vaadin-checkbox>
+          <vaadin-checkbox value="2">Option two</vaadin-checkbox>
+          <vaadin-checkbox value="3">Option three</vaadin-checkbox>
+        </vaadin-checkbox-group>
+
+        <h4>Above</h4>
+        <vaadin-checkbox-group label="Label" helper-text="Helper text above text field" theme="helper-above-field">
```
Comment (tomivirkki) [inline demo/checkbox-theme-variants-demos.html:38]: Mentions "text field". Maybe just "...above the field"?

File: test/vaadin-checkbox-group_test.html [hunk 5]
```diff
@@ -312,6 +320,69 @@
...
+        vaadinCheckboxGroup.helperText = null;
+        expect(vaadinCheckboxGroup.hasAttribute('has-helper')).to.be.false;
+      });
+
+      it('setting number helper updates has-helper attribute', function() {
+        vaadinCheckboxGroup.helperText = 0;
+        expect(vaadinCheckboxGroup.hasAttribute('has-helper')).to.be.true;
+      });
+
+      it('text-field with slotted helper updates has-helper attribute', function() {
```
Comment (tomivirkki) [inline test/vaadin-checkbox-group_test.html:360]: text-field -> field

File: demo/checkbox-basic-demos.html [hunk 6]
```diff
@@ -45,6 +45,32 @@ <h3>Disabled state</h3>
...
+      <template preserve-content>
+        <vaadin-checkbox-group label="Helper text" theme="vertical"
```
Comment (tomivirkki) [inline demo/checkbox-basic-demos.html:52]: It took me a moment to realize that the text that said "Helper text" wasn't the helper text in the demo, but the label XD Maybe the label should say something else instead.

File: src/vaadin-checkbox-group.html [hunk 7]
```diff
@@ -348,6 +392,22 @@
             this.removeAttribute('focused');
           }
         }
+
+        /** @private */
+        _setOrToggleHasHelperAttribute() {
+          const slottedNodes = this.shadowRoot.querySelector(`[name="helper"]`).assignedNodes();
+          // Only has slotted helper if not a text node
+          // Text nodes are added by the helperText prop and not the helper slot
+          // The filter is added due to shady DOM triggering this slotchange event on helperText prop change
```
Comment (tomivirkki) [inline src/vaadin-checkbox-group.html:401]: "slotchange" listener is not in use. Hmm...why not?

Comment (DiegoCardoso) [reply src/vaadin-checkbox-group.html:401]: Method is being called from the `FlattenedNodesObserver` inside `ready` method.

File: demo/checkbox-basic-demos.html (hunk 6 above)
Comment (DiegoCardoso) [reply demo/checkbox-basic-demos.html]: Changed to "Label" instead.

File: test/vaadin-checkbox-group_test.html (hunk 5 above)
Comment (DiegoCardoso) [reply test/vaadin-checkbox-group_test.html]: Done.

File: demo/checkbox-theme-variants-demos.html (hunk 4 above)
Comment (DiegoCardoso) [reply demo/checkbox-theme-variants-demos.html]: Done.

File: demo/checkbox-theme-variants-demos.html (hunk 3 above)
Comment (DiegoCardoso) [reply demo/checkbox-theme-variants-demos.html]: Done.

File: demo/checkbox-theme-variants-demos.html (hunk 2 above)
Comment (DiegoCardoso) [reply demo/checkbox-theme-variants-demos.html]: Done.

File: demo/checkbox-basic-demos.html (hunk 1 above)
Comment (DiegoCardoso) [reply demo/checkbox-basic-demos.html]: Done.

File: demo/checkbox-basic-demos.html [hunk 8]
```diff
@@ -45,6 +45,32 @@ <h3>Disabled state</h3>
       </template>
     </vaadin-demo-snippet>
 
+    <h3>Helper text</h3>
+    <p>Use the <code>helper-text</code> attribute or add content to the <code>helper</code> slot to set helper content.</p>
+    <vaadin-demo-snippet id="basic-demos-helper-text">
+      <template preserve-content>
+        <vaadin-checkbox-group label="Label" theme="vertical"
+          helper-text="Pick on option">
```
Comment (tomivirkki) [inline demo/checkbox-basic-demos.html:53]: nit: typo

Comment (DiegoCardoso) [reply demo/checkbox-basic-demos.html]: Thanks. Changed.

**Pull request merged**